### PR TITLE
[GSoC] [UI] Dashboard Page

### PR DESCRIPTION
**Pull request by Sylvie-Wxr**
### <Feature description>
- [ ] If this is non-trivial feature, paste the links/URLs to the design doc.
- [x] Update the documentation to include this new feature.
- [ ] Tests(including UT, IT, E2E) are added to verify the new feature.
- [x] If it's UI related, attach the screenshots below.
- [x] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Fixes apache/skywalking#10634.
- [ ] Update the [`CHANGES` log](https://github.com/apache/skywalking-banyandb/blob/main/CHANGES.md).

<br>

**Dashboard Page Preview:**
Dashboard default page

**Auto Refresh Seletion:**
Auto Refresh
When the page opens, the default is off, if select any option for auto refresh, the end time of datepicker will change every selected time frame, as well as the page.

**Date Picker:**
Date Picker
User can choose from different period of time as start time to current time as end time.

**Path display:**
truncate path
if the path is too long (over than 35 characters, it will show the first 5 and last 30 characters with '..." in the middle. 

**Smaller window display:**
Text not display for small screen
Time components seperate lines
When the window is sizing down, the table dynamically sizes down. When it reaches certain pixel, the text in Memory and Disk Details will not be displayed, and at a more smaller pixel, the time componenents will be in shown in seperate lines to avoid messiness.

**Table scroll with fixed time components:**
Table scroll
Make the table scroll and time components fixed so that users can still monitor or choose different time if there are more lines of nodes.

**Color display of the bar in CPU, Memory and Disk:**
Bar color
(The picture is only for display only here). In actual code, the bar will be green when CPU/Memory/Disk is less tan 50%, will be Yellow from 50%-80%, red over 80%

**No data display**:
No data page
If there is no data written in the backend, there will be no data displayed like this.

### CODE REVIEW

**Comment by hanahmily** [comment]
@Sylvie-Wxr Would you introduce the Dashboard in [observablity.md#metrics](https://github.com/apache/skywalking-banyandb/blob/main/docs/observability.md#metrics) and add brief introduction and screenshot in [`cluster.md`](https://github.com/apache/skywalking-banyandb/blob/main/docs/installation/cluster.md)

First, explain how to enable the collection of self-observability metrics. To illustrate the flow of metrics data, insert the diagram from your design into the observability. Md file. 

Next, it's important to specify the location where the metrics are stored. Lastly, ensure that the screenshot is included on the page.

**Comment by wu-sheng** [comment]
png file should be hosted in the website repo to keep codes repo volume controllable.

**Comment by wu-sheng** [comment]
Consider adding here, https://github.com/apache/skywalking-website/tree/master/static/doc-graph/banyandb/v0.7.0

Note, this is for 0.7 BanyanDB doc only.

**Comment by Sylvie-Wxr** [comment]
> Consider adding here, https://github.com/apache/skywalking-website/tree/master/static/doc-graph/banyandb/v0.7.0
> 
> Note, this is for 0.7 BanyanDB doc only.

Understood. Removed PNG from this repo.

File: docs/observability.md [hunk 1]
```diff
@@ -4,12 +4,33 @@ This document outlines the observability features of BanyanDB, which include met
...
+
+Prometheus is auto enabled at run time, if no flag is passed or if `promethus` is set in `observability-modes` flag.
 
 When the Prometheus metrics provider is enabled, the metrics server listens on port `2121`. This allows Prometheus to scrape metrics data from BanyanDB for monitoring and analysis.
 
 The Docker image is tagged as "prometheus" to facilitate cloud-native operations and simplify deployment on Kubernetes. This allows users to directly deploy the Docker image onto their Kubernetes cluster without having to rebuild it with the "prometheus" tag.
 
+### Self-observability 
+
+If the `observability-modes` flag is set to `native`, the self-observability metrics provider will be enabled. The metrics will be displayed in the dashboard of [skywalking-banyandb-webapp](http://localhost:17913/)
```
Comment (hanahmily) [inline docs/observability.md:19]:
```suggestion
If the `observability-modes` flag is set to `native`, the self-observability metrics provider will be enabled. The some of metrics will be displayed in the dashboard of [banyandb-ui](http://localhost:17913/) 
```

File: docs/observability.md [hunk 2]
```diff
@@ -4,12 +4,33 @@ This document outlines the observability features of BanyanDB, which include met
 
 ## Metrics
 
-BanyanDB has built-in support for metrics collection. Currently, there is only one supported metrics provider: `Prometheus`. It is auto enabled at run time through `observability-modes` flag. 
+BanyanDB has built-in support for metrics collection. Currently, there are two supported metrics provider: `prometheus` and `native`. These can be enabled through `observability-modes` flag.
```
Comment (hanahmily) [inline docs/observability.md:7]:
```suggestion
BanyanDB has built-in support for metrics collection. Currently, there are two supported metrics provider: `prometheus` and `native`. These can be enabled through `observability-modes` flag, allowing you to activate one or both of them. 
```

File: docs/observability.md [hunk 3]
```diff
@@ -4,12 +4,33 @@ This document outlines the observability features of BanyanDB, which include met
...
 
+### Self-observability 
+
+If the `observability-modes` flag is set to `native`, the self-observability metrics provider will be enabled. The metrics will be displayed in the dashboard of [skywalking-banyandb-webapp](http://localhost:17913/) 
+
+![dashboard](https://skywalking.apache.org/doc-graph/banyandb/v0.7.0/dashboard.png)
+
+#### Metrics storage 
+
+In self-observability, the metrics data is stored in BanyanDB within the ` _monitoring` internal group. Each metric will be created as a new `measure` within this group.
```
Comment (hanahmily) [inline docs/observability.md:25]:
```suggestion
In self-observability, the metrics data is stored in BanyanDB within the ` _monitoring` internal group. Each metric will be created as a new `measure` within this group.

You can use BanyanDB-UI or bydbctl to retrieve the data.
```

File: docs/observability.md [hunk 4]
```diff
@@ -4,12 +4,33 @@ This document outlines the observability features of BanyanDB, which include met
...
+
+#### Write Flow
+
+![self-observability-write](https://skywalking.apache.org/doc-graph/banyandb/v0.7.0/self-observability-write.png)
```
Comment (hanahmily) [inline docs/observability.md:29]: Please provide detailed comments to explain the content of this diagram

**Review by hanahmily — CHANGES_REQUESTED** [review]

File: docs/observability.md (hunk 4 above)
Comment (Sylvie-Wxr) [reply docs/observability.md:32]: Added explanation.

**Comment by hanahmily** [comment]
@Sylvie-Wxr Would you please attach the Dashboard screenshot at [cluster installation](https://github.com/apache/skywalking-banyandb/blob/main/docs/installation/cluster.md#role-base-banyand-cluster)

**Comment by Sylvie-Wxr** [comment]
> @Sylvie-Wxr Would you please attach the Dashboard screenshot at [cluster installation](https://github.com/apache/skywalking-banyandb/blob/main/docs/installation/cluster.md#role-base-banyand-cluster)

@hanahmily Updated.

**Review by hanahmily — APPROVED** [review]

**Pull request merged**